pkg/secrets/k8s_secrets_storage/provide_conjur_secrets_test.go

```diff
@@ -372,6 +375,7 @@ func TestProvide(t *testing.T) {
 		for secretName, secretData := range tc.k8sSecrets {
```

Comment:
While we're changing these unit tests, would you mind modifying the way that we're invoking test cases so that each is called via `t.Run(tc.desc ...)`. What the iterations would look like is:

```
	for _, tc := range testCases {
		t.Run(tc.desc, func(t *testing.T) {
			// Set up test case
			mocks := newTestMocks()
			. . .
```

The advantage to doing this is that if/when UT failures occur, the `tc.desc` is displayed prominently in the failure summary, and it makes it very obvious which test case(s) have failed.

Also, I believe that we would need to run the test cases via `t.Run(...)` if we should ever want to run the test cases in parallel (reference: https://eleni.blog/2019/05/11/parallel-test-execution-in-go/).